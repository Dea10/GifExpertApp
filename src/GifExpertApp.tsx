import React, { useState } from 'react';
import AddCategoryField from './components/AddCategoryField/AddCategoryField';
import GifGrid from './components/GifGrid/GifGrid';
import styles from './GifExpertApp.module.scss';

const GifExpertApp = () => {

    const [categories, setCategories] = useState<string[]>([]);

    return (
        <div>
            <h2>GifExpertApp</h2>
            <hr />

            <AddCategoryField
                setCategories={setCategories}
            />

            <div className={ styles.container }>
                {categories.map(category => (
                    <GifGrid
                        key={category}
                        category={category}
                    />
                ))}
            </div>
        </div>
    );
}

export default GifExpertApp;