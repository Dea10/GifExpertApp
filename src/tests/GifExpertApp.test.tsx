import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';


describe('GifExpertApp tests', () => {
    test('should render without errors', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });
});