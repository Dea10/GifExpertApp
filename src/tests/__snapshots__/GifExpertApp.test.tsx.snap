// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp tests should render without errors 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategoryField
    setCategories={[Function]}
  />
  <div
    className="container"
  />
</div>
`;
